### ISSUE TITLE
Regex in headers doesn't work for Response

### ISSUE THREAD

**Issue by adamwojszczyk** [Codearte/accurest#113]
For Groovy DSL:

``` groovy
io.codearte.accurest.dsl.GroovyDsl.make {
    request {
        method 'POST'
        url $(client(regex('/partners/[0-9]+/users')), server('/partners/1000/users'))
        headers { header 'Content-Type': 'application/json' }
        body(
                email: $(client(regex(email)), server('john.smith@4finance.com')),
                first_name: 'John',
                last_name: 'Smith',
                personal_id: '12345678901',
                phone_number: '500500500',
                invitation_token: '00fec7141bb94793bfe7ae1d0f39bda0',
                password: 'john'
        )
    }
    response {
        status 201
        headers {
            header 'Location': $(client('http://localhost/partners/1000/users/1001'), server(regex('http://localhost/partners/[0-9]+/users/[0-9]+')))
        }
    }
}
```

the created test doesn't recognize the Pattern and checks for equality for Response Location Header

``` groovy
def createUser() {
        given:
            def request = given()
                    .header('Content-Type', 'application/json')
                    .body('{"email":"john.smith@4finance.com","first_name":"John","last_name":"Smith","personal_id":"12345678901","phone_number":"500500500","invitation_token":"00fec7141bb94793bfe7ae1d0f39bda0","password":"john"}')

        when:
            def response = given().spec(request)
                    .post("/partners/1000/users")

        then:
            response.statusCode == 201
            response.header('Location') == 'http://localhost/partners/[0-9]+/users/[0-9]+'
    }
```

**Comment by marcingrzejszczak** [comment]
should work after release - https://github.com/Codearte/accurest/commit/daa3ccb7ea0a55f6af16c17aeac702f8d73a7222#diff-ff0c31096d76fe449d5a32fee665dd7eR333

**Comment by OlgaMaciaszek** [comment]
Implemented in 0.8.3.
